### PR TITLE
Remove org.bouncycastle dependency

### LINKED ISSUE #8362: Remove compile time dependency on bouncycastle
There are unnecessary compile time dependencies on bouncycastle. Let's remove them.

In `text-buffer`, SHA-3 is used and it seems unnecessary. Let's only use SHA-1 from JDK. Moreover, as declared in [build.sbt](https://github.com/enso-org/enso/blob/1c936cc69ff1d4e5f104209ec19bad8f8fb847fb/build.sbt#L560), `text-buffer` depends on bouncycastle JDK1.5, instead of the newer JDK1.8.

The dependency on `org.bouncacastle` in `runtime` can be removed via something like this:
```diff
diff --git a/engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java b/engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java
index 96cb560fb..b8cd239d6 100644
--- a/engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java
+++ b/engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java
@@ -10,11 +10,11 @@ import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
 import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Optional;
 import java.util.logging.Level;
-import org.bouncycastle.jcajce.provider.digest.SHA1;
 import org.bouncycastle.util.encoders.Hex;
 import org.enso.interpreter.runtime.EnsoContext;
 import org.enso.logger.masking.MaskedPath;
@@ -392,7 +392,11 @@ public abstract class Cache<T, M extends Cache.Metadata> {
    * @return digest used for computing hashes
    */
   protected MessageDigest messageDigest() {
-    return new SHA1.Digest();
+    try {
+      return MessageDigest.getInstance("SHA-1");
+    } catch (NoSuchAlgorithmException e) {
+      throw new IllegalStateException(e);
+    }
   }
 
   /**
```

Note that we need runtime dependency on `org.bouncycastle` modules because `python-language-23.1.0` depends on those anyway.

### PR DESCRIPTION
**Pull request by Akirathan**
Fixes #8362 

### Pull Request Description

Remove the `org.bouncycastle` dependency from `org.enso.runtime` module.

### Important Notes

- Digest computation is done with JDK classes instead of `org.bouncycastle`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] Manually run engine benchmarks and ensure that the performance of startup benchmarks is back to its previous value before #8351
- [x] The documentation has been updated, if necessary.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: lib/scala/text-buffer/src/main/java/org/enso/text/Hex.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.enso.text;
+
+/**
+ * A utility class that serves as a replacement of some methods from {@code org.bouncycastle}, so
+ * that we don't have to depend on it.
+ */
+public final class Hex {
```
Comment (JaroslavTulach) [inline lib/scala/text-buffer/src/main/java/org/enso/text/Hex.java:7]: Maybe you don't want to make the class `public`.

File: lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala [hunk 2]
```diff
@@ -2,14 +2,14 @@ package org.enso.text
 
 import java.nio.charset.StandardCharsets
 
-import org.bouncycastle.jcajce.provider.digest.SHA3
+import java.security.MessageDigest
```
Comment (JaroslavTulach) [inline lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala:5]: Great.

File: lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala [hunk 3]
```diff
@@ -2,14 +2,14 @@ package org.enso.text
...
 
 /** SHA3-224 digest calculator. */
 object Sha3_224VersionCalculator extends ContentBasedVersioning {
 
   /** @inheritdoc */
   override def evalVersion(content: CharSequence): ContentVersion = {
-    val digestSHA3 = new SHA3.Digest224()
+    val digestSHA3 = MessageDigest.getInstance("SHA3-224")
```
Comment (JaroslavTulach) [inline lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala:12]: Use SHA1, if possible.

File: engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java [hunk 4]
```diff
@@ -393,7 +393,11 @@ protected final String computeDigestOfLibrarySources(
    * @return digest used for computing hashes
    */
   protected MessageDigest messageDigest() {
-    return new SHA1.Digest();
+    try {
+      return MessageDigest.getInstance("SHA-1");
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java:397]: Do we know what implementation of SHA-1 algorithm we get? JDK one or bouncycastle one?

**Review by JaroslavTulach — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

File: lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala (hunk 3 above)
Comment (Akirathan) [reply lib/scala/text-buffer/src/main/scala/org/enso/text/Sha3_224VersionCalculator.scala:12]: Tracked as a separate issue in https://github.com/enso-org/enso/issues/8662

**Comment by Akirathan** [comment]
Engine benchmarks scheduled at https://github.com/enso-org/enso/actions/runs/7400828290

stdlib benchmarks scheduled at https://github.com/enso-org/enso/actions/runs/7400834645

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Engine · enso-org/enso@b85d5e5</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Standard Libraries · enso-org/enso@b85d5e5</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Review by jdunkerley — APPROVED** [review]

**Review by GregoryTravis — APPROVED** [review]

**Review by radeusgd — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
> stdlib benchmarks scheduled at https://github.com/enso-org/enso/actions/runs/7400834645

There is a [failure related to bouncycastle](https://github.com/enso-org/enso/actions/runs/7400834645/job/20135312051#step:10:938):
```
running (fork) org.enso.benchmarks.libs.LibBenchRunner 
Error occurred during initialization of boot layer
java.lang.LayerInstantiationException: Package org.bouncycastle.tsp.ers in both module 
  org.enso.runtime and module org.bouncycastle.pkix
 (std-benchmarks / Bench / run) Nonzero exit code returned from runner: 1
```

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Standard Libraries · enso-org/enso@b85d5e5</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

File: lib/scala/text-buffer/src/main/java/org/enso/text/Hex.java (hunk 1 above)
Comment (Akirathan) [reply lib/scala/text-buffer/src/main/java/org/enso/text/Hex.java:7]: It is used both from within `text-buffer` and also from `runtime`. It is sort of a minified replacement of `org.bouncycastle.Hex`.

File: engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java (hunk 4 above)
Comment (Akirathan) [reply engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java:397]: It's Sun provider, so the JDK one.

**Comment by Akirathan** [comment]
> There is a [failure related to bouncycastle](https://github.com/enso-org/enso/actions/runs/7400834645/job/20135312051#step:10:938):

This is because of the incremental builds. Let me clean all the runners and reschedule that job again.

**Comment by Akirathan** [comment]
Another attempt of std-lib benchmarks started at https://github.com/enso-org/enso/actions/runs/7408255173

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Standard Libraries · enso-org/enso@b85d5e5</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by Akirathan** [comment]
> Another attempt of std-lib benchmarks started at [enso-org/enso/actions/runs/7408255173](https://github.com/enso-org/enso/actions/runs/7408255173)
> 
> >  GitHub**[Benchmark Standard Libraries · enso-org/enso@b85d5e5](https://github.com/enso-org/enso)**Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.

Looking at the bench comparison at [generated_site.zip](https://github.com/enso-org/enso/files/13832559/generated_site.zip), removing `org.bouncycastle` dependency from `org.enso.runtime` does not help in `org_enso_benchmarks_generated_Startup_empty_startup`.

Engine benchmarks are the same.

**Comment by Akirathan** [comment]
Although removing `org.bouncycastle` dependency did not help with the benchmarks, it is still valuable. Let's merge this PR now, and continue with the startup slowdown investigation in another PR.

**Pull request merged**